### PR TITLE
Add headers + CRUD using frontend  + access remote graphiti servers

### PR DESCRIPTION
**Pull request by XHPJ**
Hello, 

This PR adds the following functionalities to vandal:

- add custom/arbitrary headers when requesting sending a request to graphiti backend  
- CRUD using the UI, relationships need to be created manually. E.g. , when a store has multiple products you would need to first create a store and then create the products and then map them via foreign keys.
- Access remote graphiti backends, when specifying a environment variable `REMOTE_HOSTS` the vandal_UI project will add these hosts to vandal and then one is able to connect to different hosts, rather than the default/own server.

The first to features also work for remote hosts. 

### Headers

The headers will be used when firing a request to the server and the fields are stored using localStorage and ls-secure for encryption. 

### CRUD

This is mostly affecting the `query.ts`. Still the payload (update/create) is passed from All.vue to the query object.

### Remote Hosts
Vandal would  expect an ENV called REMOTE_HOST which contains a JSON array with the following objects:
```
{ 
"name" : "<application name that will be shown in frontend>",
"url" : "https://<server url>"},    ## 'https://' needs to be there 
"schemaPath" : "</path/to/schema>/schema.json"
}
```

Right now vandal_ui looks inside `index.html` and replaced the string `__REMOTE_HOSTS__` with the value of the ENV or an `{}` when no value is set.

Side note: I will create a PR for vandal_ui with the same branch name, that uses this version of vandal.

### CODE REVIEW

**Comment by XHPJ** [comment]
Note: I have also a little screen recording prepared in case you want to see this for easier evaluation, I can sent it to you via Slack.

**Comment by richmolj** [comment]
> Note: I have also a little screen recording prepared in case you want to see this for easier evaluation, I can sent it to you via Slack.

I would love to see this, thank you!

Would you mind running a `git rebase -i` to consolidate these commits?

File: package.json [hunk 1]
```diff
@@ -12,11 +12,15 @@
   "dependencies": {
     "@types/highlight.js": "9.12.3",
     "@types/moment": "2.13.0",
+    "axios": "^0.21.0",
```
Comment (richmolj) [inline package.json]: Was it not possible to avoid this dependency?

File: .prettierrc.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Comment (richmolj) [inline .prettierrc.js]: While I'm happy to add prettier and linting rules to this repo, ideally it would be in a separate PR (makes everything easier to review)

Comment (XHPJ) [reply .prettierrc.js]: yeah right, I added this to my branch because my IDE was changing way to much, I think I will integrate this in the .gitignore

File: package.json (hunk 1 above)
Comment (XHPJ) [reply package.json]: seems like I forgot to remove this, will do this
